setup.json

```diff
@@ -68,8 +68,7 @@
     "extras_require": {
         "pre-commit": [
             "pre-commit~=2.2",
-            "prospector~=1.2",
-            "yapf~=0.29"
+            "pylint~=2.5"
```

Comment:
Here, `pylint` also needs to be at a specific version (I guess ``~=2.5.0`` would also be ok, allowing bugfix-release changes).

The reason for this is that the ``language`` specified above is ``system``: This means that pylint won't actually be installed from the source specified in ``.pre-commit-config.yaml``, only the _hook_ specification from there is used. Instead, it uses the system-installed (or usually: virtualenv-installed) ``pylint``. You can check this by intentionally breaking ``.virtualenvs/<venv_name>/bin/pylint``.

The reason we actually _need_ to use the ``system`` language is because ``pylint`` needs access to all the other packages in the virtualenv to do its analysis properly.

Reply:
You don't wanna know how long it took me to figure this out the first time around 😅

Reply:
Ah yes, I also broke my head over this and then forgot why I also needed to specify it in the `extras_require`. It really is a shitty solution. Do you know if this is a problem of `pylint` or just of `pre-commit` and so there will be no away around it?

Reply:
I think that's a requirement of ``pylint``.. it's even mentioned in the docs of ``pre-commit`` as to why one might use the ``system`` language: https://pre-commit.com/#system

Why ``pylint`` itself has ``python`` as a language in its hook definition is beyond me.